frontend/src/app/settings/modelsources/t4c-settings/licences/licences.component.ts

```diff
@@ -3,25 +3,38 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { Component, OnInit } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
+import { Component, OnInit, Input } from '@angular/core';
 import { SessionUsage } from 'src/app/schemes';
-import { SessionService } from 'src/app/services/session/session.service';
+import {
+  T4CInstance,
+  T4CInstanceService,
+} from 'src/app/services/settings/t4c-instance.service';
 
 @Component({
   selector: 'app-licences',
   templateUrl: './licences.component.html',
   styleUrls: ['./licences.component.css'],
 })
 export class LicencesComponent implements OnInit {
-  constructor(private sessionService: SessionService) {}
+  @Input() instance!: T4CInstance;
+
+  public errorMessage?: string;
+
+  constructor(private t4cInstanceService: T4CInstanceService) {}
 
   ngOnInit(): void {
-    this.sessionService.getSessionUsage().subscribe((res) => {
-      this.sessionUsage = res;
+    this.t4cInstanceService.getLicenses(this.instance.id).subscribe({
+      next: (res) => {
+        this.sessionUsage = res;
+      },
+      error: (err: HttpErrorResponse) => {
+        this.errorMessage = this.getErrorMessage(err.error.detail.code);
+      },
     });
   }
 
-  sessionUsage: SessionUsage | undefined = undefined;
+  sessionUsage?: SessionUsage;
 
   getErrorMessage(error: string): string {
```

Comment:
This method is not needed anymore. We can just add the message as error message ("reason") in the backend and use the existing backend messages as "technical". Then we don't have the duplication of error messages.

Reply:
You now just removed it but the information was not added to the error messages? 
E.g. "This can happen during and after license server restarts." is really important and should be displayed to the user.

Reply:
Not resolved yet as far as I can see.

Reply:
I changed the base to itself and the solution is now visible on GitHub.

Reply:
Solved.